x/oracle/keeper/query_get_reports.go

```diff
@@ -66,52 +66,72 @@ func (k Querier) GetReportsbyQid(ctx context.Context, req *types.QueryGetReports
 func (k Querier) GetReportsbyReporter(ctx context.Context, req *types.QueryGetReportsbyReporterRequest) (*types.QueryMicroReportsResponse, error) {
 	if req == nil {
 		return nil, status.Error(codes.InvalidArgument, "invalid request")
 	}
 
-	reporter := sdk.MustAccAddressFromBech32(req.Reporter)
-	pageRes := &query.PageResponse{
-		NextKey: nil,
-		Total:   uint64(0),
+	reporter, err := sdk.AccAddressFromBech32(req.Reporter)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid reporter address")
 	}
 
-	// key is Bytes (reporter address) with bytes encoded max uint64 concatenated (reporterAddr...fff...)
-	buffer := make([]byte, 8)
-	_, err := collections.Uint64Key.Encode(buffer, ^uint64(0))
-	if err != nil {
-		return nil, status.Error(codes.Internal, "failed to encode start value")
+	limit := uint64(10)
+	if req.Pagination != nil && req.Pagination.Limit > 0 {
+		limit = req.Pagination.Limit
 	}
 
-	// construct key: reporter_address + encoded_uint64
-	key := append(reporter.Bytes(), buffer...)
-	rng := collections.NewPrefixUntilPairRange[[]byte, collections.Triple[[]byte, []byte, uint64]](key)
+	// Create custom range for this reporter
+	reporterRange := types.NewReporterRange(reporter.Bytes())
+
+	// Apply ordering
 	if req.Pagination != nil && req.Pagination.Reverse {
-		rng.Descending()
+		reporterRange = reporterRange.Descending()
 	}
-	tripleKeyCodec := collections.TripleKeyCodec(collections.BytesKey, collections.BytesKey, collections.Uint64Key)
+
+	// Handle pagination start key
 	if req.Pagination != nil && len(req.Pagination.Key) > 0 {
+		tripleKeyCodec := collections.TripleKeyCodec(collections.BytesKey, collections.BytesKey, collections.Uint64Key)
 		_, startKey, err := tripleKeyCodec.Decode(req.Pagination.Key)
 		if err != nil {
 			return nil, status.Error(codes.InvalidArgument, "invalid pagination key")
 		}
-		rng.StartInclusive(startKey)
+		// Convert the primary key to an index pair for range starting
+		// The index key is constructed from reporter address + metaId
+		size := collections.Uint64Key.Size(startKey.K3())
+		buffer := make([]byte, size)
+		_, err = collections.Uint64Key.Encode(buffer, startKey.K3())
+		if err != nil {
+			return nil, status.Error(codes.Internal, "failed to encode metaId")
+		}
+		indexKey := append(startKey.K2(), buffer...)
+		startPair := collections.Join(indexKey, startKey)
+		reporterRange = reporterRange.StartInclusive(startPair)
 	}
 
-	iter, err := k.keeper.Reports.Indexes.Reporter.Iterate(ctx, rng)
+	// Use the custom range with the Reporter index
+	iter, err := k.keeper.Reports.Indexes.Reporter.Iterate(ctx, reporterRange)
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
+	defer iter.Close()
+
 	reports := make([]types.MicroReportStrings, 0)
+	pageRes := &query.PageResponse{
+		NextKey: nil,
+		Total:   uint64(0),
+	}
+
 	for ; iter.Valid(); iter.Next() {
```

Comment:
something like this could get the `--page-offset` flag working.
```
	// Skip offset number of results
	offsetCount := uint64(0)
	if req.Pagination != nil {
		offsetCount = req.Pagination.Offset
	}

	skipped := uint64(0)
	for ; iter.Valid(); iter.Next() {
		// Skip offset number of results first
		if skipped < offsetCount {
			skipped++
			continue
		}
```

Reply:
from the sdk:
```go
if !advanceIter(iterator, offset) {
		return nil, nil, collections.ErrInvalidIterator
	}
	
func advanceIter[I interface {
	Next()
	Valid() bool
}](iter I, offset uint64,
) bool {
	for i := uint64(0); i < offset; i++ {
		if !iter.Valid() {
			return false
		}
		iter.Next()
	}
	return true
}
```